### PR TITLE
#7 Add Support object literal arguments

### PR DESCRIPTION
**Pull request by Allali84**
Adding the support for the object literal arguments

### CODE REVIEW

File: src/main/kotlin/com/apurebase/kgraphql/schema/execution/ArgumentTransformer.kt [hunk 1]
```diff
@@ -66,4 +68,36 @@ open class ArgumentTransformer(val schema : DefaultSchema) {
     fun transformPropertyValue(parameter: InputValue<*>, value: String, variables: Variables): Any? {
         return transformValue(parameter.type, value, variables)
     }
+
+    fun transformPropertyObjectValue(parameter: InputValue<*>, value: List<*>): Any? {
+        val mapper = ObjectMapper()
+        mapper.registerModule(JavaTimeModule())
```
Comment (jeggy) [inline src/main/kotlin/com/apurebase/kgraphql/schema/execution/ArgumentTransformer.kt]: What's the reasoning behind adding this module?

**Comment by jeggy** [comment]
This looks great! 

Thanks for your help on this 😃

File: src/main/kotlin/com/apurebase/kgraphql/schema/execution/ArgumentTransformer.kt (hunk 1 above)
Comment (Allali84) [reply src/main/kotlin/com/apurebase/kgraphql/schema/execution/ArgumentTransformer.kt]: This module is neaded for the support of Java 8 date/time API in the FasterXML framework.
In my project i used the "java.time.LocalDate" and i got this exception : `"com.fasterxml.jackson.databind.exc.InvalidDefinitionException:  Cannot construct instance of java.time.LocalDate`
The solution was adding this module

Comment (jeggy) [reply src/main/kotlin/com/apurebase/kgraphql/schema/execution/ArgumentTransformer.kt]: Wouldn't this work instead:
```kotlin
fun transformPropertyObjectValue(parameter: InputValue<*>, value: List<*>): Any? {
    return schema.configuration.objectMapper.readValue(value.toJson(), parameter.type.unwrapped().kClass?.java)
}
```

And then you would create a custom scalar for the `java.time.LocalDate` type?

Comment (Allali84) [reply src/main/kotlin/com/apurebase/kgraphql/schema/execution/ArgumentTransformer.kt]: Worked perfectly 👍 
I will modify the code and commit it right away 😃

**Comment by jeggy** [comment]
Thanks for this, I myself have been waiting for this feature 😄 

It's being published right now and should be available within a few minutes on bintray 👌

**Comment by Allali84** [comment]
I'm glad i could help 😄
This is a cool API and i'm happy that i contributed on it

**Pull request merged**